test/puppetlabs/trapperkeeper/services_test.clj

```diff
@@ -62,6 +63,55 @@
           app (bootstrap-services-with-empty-config [service1])]
       (is (not (nil? app))))))
 
+(deftest test-restart-file-correct-count
+  (testing "create a restart file and check that it correctly increments when a service starts"
+    (let [service1 (service Service1
+                            []
+                            (service1-fn [this] "hi"))
+          service2 (service Service1
+                            []
+                            (service1-fn [this] "hi"))]
+          (with-app-with-config app [service1] {:global {:restart-file "./dev-resources/counter"}}
```

Comment:
If you add the function to the `a/restart` lifecycle function, I would remove the second service from this test and just call restart on the app, and then assert that the count had been incremented.